drivers/pinctrl/pinctrl_nrf.c

```diff
@@ -319,7 +321,9 @@ int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt,
 		}
 
 		/* configure GPIO properties */
-		if (pin != NRF_PIN_DISCONNECTED) {
```

Comment:
This PR could just be one line change: `pin != 0xFFFFFFFFU`, or 2 lines if using `#define PSEL_DISCONNECTED 0xFFFFFFFFUL`. I think there's no need to change naming, what we are writing to the PSEL (Pin SELect) register is a pin number in the end.

Reply:
> what we are writing to the PSEL (Pin SELect) register is a pin number in the end.

Unless a given signal is not be connected to a pin. Then the value is not a pin number.
I propose this name change exactly to avoid such confusion.